near/src/runtime.rs

```diff
@@ -1137,7 +1141,8 @@ impl node_runtime::adapter::ViewRuntimeAdapter for NightshadeRuntime {
                 })
                 .collect::<Result<Vec<_>, Box<dyn std::error::Error>>>(),
             Err(e) => Err(e.into()),
-        }
+        };
+        access_keys
```

Comment:
For some reason, Rust fails to manage lifetimes here and I am forced to save the intermediate result in a temporary variable to avoid:

```
error[E0597]: `state_update` does not live long enough
    --> near/src/runtime.rs:1131:15
     |
1131 |         match state_update.iter(&prefix) {
     |               ^^^^^^^^^^^^--------------
     |               |
     |               borrowed value does not live long enough
     |               a temporary with access to the borrow is created here ...
...
1145 |     }
     |     -
     |     |
     |     `state_update` dropped here while still borrowed
     |     ... and the borrow might be used here, when that temporary is dropped and runs the destructor for type `std::result::Result<near_store::trie::update::TrieUpdateIterator<'_>, near_primitives::errors::StorageError>`
     |
     = note: The temporary is part of an expression at the end of a block. Consider forcing this temporary to be dropped sooner, before the block's local variables are dropped. For example, you could save the expression's value in a new local variable `x` and then make `x` be the expression at the end of the block.
```
```

Reply:
Rust compilation errors are awesome, they literally tells us what happened, why, and how to fix it.
The problem is that method returns result after dropping everything that was created in the function scope. So it trie to return the result of `match state_update.iter(&prefix) {...}` expression but the problem is that because it is an expression it still holds all the references that it used including reference to `state_update`, but `state_update` is already dropped and so it complains. The root problem is that in Rust lifetimes are not smart when they are grabbed by the expression. Lifetimes are only smart within a block across various statements (statements are composed of expressions). Basically lifetime smartness stops at the level of statements and does not act on expressions.